src/java.base/share/classes/java/util/Formatter.java

```diff
@@ -3823,6 +3823,7 @@ else if (precision == 0)
 
                 BigDecimal tenToTheNegFour = BigDecimal.valueOf(1, 4);
                 BigDecimal tenToThePrec = BigDecimal.valueOf(1, -prec);
+                value = value.round(new MathContext(prec));
```

Comment:
While you are in the area, how about inlining the creation of the tenToTheNegFour and tenToThePrec values.
They are used only once and may not be used at all. They don't appear to be needed except for the comparisons.

Reply:
Makes sense to me.